features/step_definitions/comment_steps.rb

```diff
@@ -19,6 +19,11 @@
   FactoryBot.create(:comment, :by_guest, commentable: work.first_chapter)
 end
 
+Given "a guest comment on the admin post {string}" do |title|
+  admin_post = AdminPost.find_by(title: title)
+  FactoryBot.create(:comment, :by_guest, commentable: admin_post)
+end
```

Comment:
I think you can merge this with the previous step definition to get a single `"a guest comment on {commentable}"` step, like the ones defined below but for guests.